src/snowflake/cli/_plugins/nativeapp/entities/application.py

```diff
@@ -765,50 +765,53 @@ def drop_application_before_upgrade(
         interactive: bool,
         cascade: bool = False,
     ):
-        if cascade:
-            try:
-                if application_objects := self.get_objects_owned_by_application():
-                    application_objects_str = _application_objects_to_str(
-                        application_objects
+        sql_executor = get_sql_executor()
+        with sql_executor.use_role(self.role):
```

Comment:
Just adding this line at the top and indenting the rest of the function. I believe this doesn't need a warehouse from my tests so I did not add it, but please let me know if I am mistaken